satpy/resample.py

```diff
@@ -615,7 +615,10 @@ def prepare_resampler(source_area, destination_area, resampler=None):
         LOG.info("Using default KDTree resampler")
         resampler = 'kd_tree'
 
-    if isinstance(resampler, str):
+    if isinstance(resampler, BaseResampler):
+        raise ValueError("Trying to create a resampler when one already "
+                         "exists.")
```

Comment:
could we just return `resampler` in this case ?

Reply:
I thought about that and that's what I originally had, but then realized I was just allowing something that never should have happened. I don't see a use case where a user would provide an existing resampler instance to the `resample` method. A resampler class, sure, a resampler instance, no.

Reply:
fine by me then